trogon/trogon.py

```diff
@@ -217,24 +210,47 @@ class Trogon(App):
 
     def __init__(
         self,
-        cli: click.Group,
-        app_name: str | None = None,
-        command_name: str = "tui",
-        click_context: click.Context | None = None,
+        command_schemas: dict[CommandName, CommandSchema],
+        app_name: str | None,
+        app_version: str | None = None,
     ) -> None:
         super().__init__()
-        self.cli = cli
         self.post_run_command: list[str] = []
-        self.is_grouped_cli = isinstance(cli, click.Group)
+        self.command_schemas = command_schemas
+        self.is_grouped_cli = any(v.subcommands for v in command_schemas.values())
         self.execute_on_exit = False
-        if app_name is None and click_context is not None:
-            self.app_name = detect_run_string()
-        else:
-            self.app_name = app_name
-        self.command_name = command_name
+
+        self.app_name = app_name if app_name else detect_run_string()
+        self.app_version = app_version
+
+        if not self.app_version:
+            with suppress(Exception):
+                self.app_version = metadata.version(self.app_name)
+
+    @classmethod
+    def from_schemas(cls, *args: CommandSchema, **kwargs) -> "Trogon":
+        if not args:
+            raise ValueError("No schemas provided.")
+
+        root_schema = args[0]
+
+        schemas: dict[CommandName, CommandSchema] = {root_schema.name: root_schema}
+
+        for schema in args[1:]:
+            schema.parent = root_schema
+            root_schema.subcommands[schema.name] = schema
+
+        return cls(schemas, **kwargs)
 
     def on_mount(self):
-        self.push_screen(CommandBuilder(self.cli, self.app_name, self.command_name))
+        self.push_screen(
+            CommandBuilder(
+                self.command_schemas,
+                app_name=self.app_name,
+                app_version=self.app_version,
+                is_grouped_cli=self.is_grouped_cli,
+            ),
+        )
```

Comment:
`CommandBuilder` now accepts the schemas directly, rather than inferring schemas from a given CLI. It is the responsibility of the various "introspectors" to do the reverse-engineering and provide schemas to `CommandBuilder`.